src/serialize.h

```diff
@@ -339,13 +340,13 @@ uint64_t ReadCompactSize(Stream& is, bool range_check = true)
  * negative numbers in a backwards compatible way, and additional modes could be
  * added to support different varint formats (e.g. zigzag encoding).
  */
-enum class VarIntMode { DEFAULT, NONNEGATIVE_SIGNED };
+enum class VarIntMode { DEFAULT, NONNEGATIVE_SIGNED, BITSTREAM };
 
 template <VarIntMode Mode, typename I>
 struct CheckVarIntMode {
     constexpr CheckVarIntMode()
     {
-        static_assert(Mode != VarIntMode::DEFAULT || std::is_unsigned<I>::value, "Unsigned type required with mode DEFAULT.");
+        static_assert(Mode != VarIntMode::DEFAULT || Mode != VarIntMode::BITSTREAM || std::is_unsigned<I>::value, "Unsigned type required with mode DEFAULT||BITSTREAM.");
```

Comment:
In ec2b2ea56c06bf72e53b0ecfb4065ccf2bd09cb0:

This boolean is wrong. I'd split it into two lines, which would be clearer (and harder to get wrong).

Reply:
Fixed, Thanks!